pyccel/codegen/printing/fcode.py

```diff
@@ -2612,10 +2613,10 @@ def _print_LiteralInteger(self, expr):
 
     def _print_IndexedElement(self, expr):
         base = expr.base
-        if isinstance(base, Application) and not isinstance(base, PythonTuple):
+        if isinstance(base, PyccelInternalFunction) and not isinstance(base, PythonTuple):
```

Comment:
Can `base` be an instance of both `PyccelInternalFunction` and `PythonTuple`?

Reply:
Not at the moment, but I think that it will when the `Expr` inheritance is removed

Reply:
I cannot imagine a class that inherits from both `PyccelInternalFunction` and `PythonTuple`. Could you give me an example?

Reply:
`tuple(1,2)`
which is translated to a `PythonTuple` but is a function

Reply:
But this is not another class that inherits from both `PyccelInternalFunction` and `PythonTuple`.

The `if` condition above is `True` if `expr` is a `PyccelInternalFunction` **and not** a `PythonTuple`, but the second condition is redundant because always true when the first is true...

Instead, this should just read
```
if isinstance(base, PyccelInternalFunction):
```

Reply:
Unless `PythonTuple` is a subclass of `PyccelInternalFunction` of course... is that the case?

Reply:
Not at the moment, but I think that it will when the `Expr` inheritance is removed

Currently `PythonTuple` inherits from `PyccelAstNode` and `Expr`. `Expr` handles the `*args`, either we could re-implement that in `PythonTuple` or we could replace the `Expr` inheritance with a `PyccelInternalFunction` inheritance

Reply:
Thanks for explaining